### ISSUE TITLE
Logical OR Interpreted as union of literal values in some cases for `Final`

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#8417]
Using `Final` with a logical OR that can possibly be statically evaluated leads to a `Literal` with each item being included in the union.

```python
from typing import Final, reveal_type

CONST_NUM: Final = 2 or 4

reveal_type(CONST_NUM) # Type of "CONST_NUM" is "Literal[2, 4]"
```

Interestingly enough, this does work correctly when the first operand is 0 or `False`.

```python
from typing import Final, reveal_type

CONST_NUM: Final = 0 or 1

reveal_type(CONST_NUM) # Type of "CONST_NUM" is "Literal[1]
```

**Comment by erictraut** [comment]
Yeah, I agree this is inconsistent. This will be addressed in the next release. The fix also applies to `and` operators and `str` and `bytes` literals in addition to `bool` and `int`.

**Comment by max-muoto** [comment]
> Yeah, I agree this is inconsistent. This will be addressed in the next release. The fix also applies to `and` operators and `str` and `bytes` literals in addition to `bool` and `int`.

Thanks!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.372.
